src/Rules.Framework/Extensions/GenericRuleExtensions.cs

```diff
@@ -0,0 +1,59 @@
+namespace Rules.Framework.Extensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Rules.Framework.Core;
+    using Rules.Framework.Core.ConditionNodes;
+    using Rules.Framework.Generics;
+
+    internal static class GenericRuleExtensions
+    {
+        public static GenericConditionNode ToGenericConditionNode<TConditionType>(this IConditionNode<TConditionType> rootCondition)
+        {
+            if (rootCondition.LogicalOperator == LogicalOperators.Eval)
+            {
+                var condition = rootCondition as ValueConditionNode<TConditionType>;
+
+                var conditionAsEnum = Enum.Parse(typeof(TConditionType), condition.ConditionType.ToString());
+
+                return new GenericValueConditionNode
+                {
+                    ConditionTypeName = conditionAsEnum.ToString(),
+                    DataType = condition.DataType,
+                    Operand = condition.Operand,
+                    Operator = condition.Operator
+                };
+            }
+
+            var composedConditionNode = rootCondition as ComposedConditionNode<TConditionType>;
+
+            var conditionNodeDataModels = new List<GenericConditionNode>(composedConditionNode.ChildConditionNodes.Count());
+
+            foreach (IConditionNode<TConditionType> child in composedConditionNode.ChildConditionNodes
+                .Where(condition => condition != null))
```

Comment:
Done

Reply:
If you want, you can add a fluent validation rule [here](https://github.com/Farfetch/rules-framework/blob/master/src/Rules.Framework/Builder/Validation/ComposedConditionNodeValidator.cs#L16) to ensure all child condition nodes are not null, that way you can be confident that a child condition node is never null in this piece of code you fixed.

Reply:
added the fluent validation :) makes all sense

Reply:
Thank you Carlos 💪

Reply:
Suggestion: Use var instead of IConditionNode<TConditionType>.

On that topic, @luispfgarces any special reason for mostly specifying the variable types throughout the code instead of using var?

I.M.O. explicitly specifying the types makes the code less easier to read and less flexible for changes.. 
As it says [here](https://intellitect.com/blog/when-to-use-and-not-use-var-in-c/#:~:text=I%20use%20var%20any%20time%20that%20the%20initialization%20of%20the%20variable%20clearly%20tells%20me%20what%20the%20variable%20will%20contain.), I believe it is only useful to specify it when it is not clear what we're assigning to a variable.

Reply:
(And, of course, if we want the variable to be of a super type or an interface of the object that we're assigning to the variable.)

Reply:
I have nothing against it, it may make the code review a bit more challenging outside of an IDE in some specific cases (like this one we commented, where you can't see the type `IConditionNode<TConditionType>`), but if you know the code well enough, that won't be an issue. Let's call this an old habit 😄 I'm all OK with setting that as a guideline for the project 👍

Reply:
Just as a side note, I am already practicing the use of var on my contributions, I think it is just a matter of setting a guideline and cleanup code accordingly.

Reply:
Understand :) I think that even if you don't know the code well enough, or don't have access to the the underlying type via the IDE, good naming of classes, methods and variables is what mostly helps developers to read and understand code.

I don't think we need to deliberately change all variables to var in the code, but we can instead apply those changes as we come across each file in our developments.

As for the guidelines, don't really know if it is justified to add a reference to this topic, given that most people already make use of var _by default_. However, if you believe it is helpful enough to add it, I can take care of it. Would that be in the Contributing.MD file?

Reply:
Yes, that would be spot, and go on, open a pull request to add that guideline there 😉

Reply:
MR created: https://github.com/Farfetch/rules-framework/pull/103